openhands/events/observation/commands.py

```diff
@@ -28,7 +145,21 @@ def success(self) -> bool:
         return not self.error
 
     def __str__(self) -> str:
-        return f'**CmdOutputObservation (source={self.source}, exit code={self.exit_code})**\n{self.content}'
+        return (
+            f'**CmdOutputObservation (source={self.source}, exit code={self.exit_code}, '
+            f'metadata={json.dumps(self.metadata.model_dump(), indent=2)})**\n'
+            '--BEGIN AGENT OBSERVATION--\n'
+            f'{self.to_agent_observation()}\n'
+            '--END AGENT OBSERVATION--'
+        )
+
+    def to_agent_observation(self) -> str:
```

Comment:
```suggestion
    def _to_agent_observation(self) -> str:
```
This method seems used only here? Maybe we can make it private.

It confused me a bit, I thought it was used in the agent... 😅